packages/cli/src/commands/transfer/celo.ts

```diff
@@ -23,41 +26,86 @@ export default class TransferCelo extends BaseCommand {
 
   async run() {
     const kit = await this.getKit()
+    const client = await this.getPublicClient()
+    const wallet = await this.getWalletClient()
     const res = await this.parse(TransferCelo)
 
+    if (!wallet) {
+      throw new Error('TODO: only AKV doesnt return a wallet')
+    }
+
     const from = res.flags.from
     const to = res.flags.to
-    const value = new BigNumber(res.flags.value)
+    const value = res.flags.value
+
+    const params =
+      // TODO: get rid of kit here
+      kit.connection.defaultFeeCurrency ? { feeCurrency: kit.connection.defaultFeeCurrency } : {}
 
-    kit.defaultAccount = from
-    const celoToken = await kit.contracts.getGoldToken()
+    const celoContract = {
+      address: await resolveAddress(client, 'GoldToken'),
+      abi: goldTokenABI,
+      account: wallet.account,
+      ...params,
+    } as const
+
+    const transferParams = { to, value: value, ...params } as const
 
     await newCheckBuilder(this)
       .isNotSanctioned(from)
       .isNotSanctioned(to)
+      .isValidWalletSigner(from)
       .hasEnoughCelo(from, value)
-      .runChecks()
+      .addCheck(
+        `Account can afford to transfer CELO with gas paid in ${
+          kit.connection.defaultFeeCurrency || 'CELO'
+        }`,
+        async () => {
+          const tokenForGasContract = kit.connection.defaultFeeCurrency
+            ? getERC20Contract
+            : getGoldTokenContract
 
-    const params = await kit.connection.setFeeMarketGas(
-      kit.connection.defaultFeeCurrency ? { feeCurrency: kit.connection.defaultFeeCurrency } : {}
-    )
+          const [gas, gasPrice, balanceOfTokenForGas] = await Promise.all([
+            res.flags.comment
+              ? client.estimateContractGas({
+                  ...celoContract,
+                  functionName: 'transferWithComment',
+                  args: [to, value, res.flags.comment],
+                })
+              : client.estimateGas(transferParams),
+            client.getGasPrice(),
+            tokenForGasContract(client as PublicClient, kit.connection.defaultFeeCurrency!).then(
+              (contract) => contract.read.balanceOf([from])
+            ),
+          ])
+          return balanceOfTokenForGas >= gas * gasPrice
```

Comment:
maybe im missreading but i think that this doesnt cover when you are paying for gas with Celo and sending celo and the combined amount is more than you are sending.

Reply:
you're correct! adding a test case for it

Reply:
done